src/clinical/service-worker-thread/tasks.ts

```diff
@@ -124,10 +125,13 @@ const sortDocs = (sort: string, entityName: string, completionStats: CompletionR
     const completionSort = completionA === completionB ? 0 : completionA > completionB ? -1 : 1;
     order = completionSort;
   } else {
+    const currentEntry = currentRecord as ClinicalInfo;
+    const nextEntry = nextRecord as ClinicalInfo;
+
     // Sort by Selected Column
     const key = isDescending === -1 ? sort.split('-')[1] : sort;
-    const first: any = currentRecord[key];
-    const next: any = nextRecord[key];
+    const first: any = currentEntry[key] as string;
+    const next: any = nextEntry[key];
```

Comment:
do these need to by typed as `any`? would prefer strict types.

Reply:
just checked this. we should not declare these as any, we know their type from context. They are being checked as `first > next` later which could have weird effects if they are an unexpected type.

Reply:
I see now. This was definitely a band aid for cascading type issues. I've updated the logic with better types, please take a look!